packages/drivers/odsp-driver/src/odspDriverUrlResolverForShareLink.ts

```diff
@@ -74,12 +69,12 @@ export class OdspDriverUrlResolverForShareLink implements IUrlResolver {
         private readonly appName?: string,
     ) {
         this.logger = createOdspLogger(logger);
-        if (shareLinkFetcherProps) {
```

Comment:
Now that you're not doing the transformation of the tokenFetcher here, I think you can delete this if block altogether and bring the field declaration into the constructor param like this:

```
public constructor(
    private readonly shareLinkFetcherProps: ShareLinkFetcherProps | undefined,
```

Checking my work:  If `shareLinkFetcherProps` is undefined, we would have done nothing, no change.  If it is defined, the object spread/reconstruction has no effect - besides cloning the object which I imagine is not necessary but you should double check.

Reply:
I can easily remove it - I did not investigate further but assumed there will be a scenario where we would like to override an existing tokenFetcher implementation.

Reply:
I would remove it but I'm not worried about it if you don't